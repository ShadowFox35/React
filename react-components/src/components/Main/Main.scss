.main {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.title {
  margin-bottom: 30px;
  font-family: 'Montserrat', sans-serif;
  font-style: normal;
  font-weight: 600;
  font-size: 24px;
  line-height: 17px;
}
