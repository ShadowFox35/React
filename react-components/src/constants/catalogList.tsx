import { catalogElemType } from 'types/objects';

export const catalogList: catalogElemType[] = [
  {
    url: 'assets/catalogItems/Zefirka.svg',
    category: 'Knitted yarn',
    name: 'Zefirka',
    brand: 'Zefirka',
    structure: '100%  cotton',
    weight: '380 g',
    length: '100 m',
    price: 17,
  },

  {
    url: 'assets/catalogItems/Arachna.svg',
    category: 'Knitted yarn',
    name: 'Sling Print',
    brand: 'Arachna',
    structure: '100%  cotton',
    weight: '160 g',
    length: '50 m',
    price: 16,
  },

  {
    url: 'assets/catalogItems/Katia_Merino_Flame.svg',
    category: 'Knitted yarn',
    name: 'Merino Flame',
    brand: 'Katia',
    structure: '74% merino, 26%  cotton',
    weight: '50 g',
    length: '80 m',
    price: 10,
  },

  {
    url: 'assets/catalogItems/Archna_Macrame_MAXI.svg',
    category: 'Knitted yarn',
    name: 'Macrame MAXI',
    brand: 'Arachna',
    structure: '74% merino, 26%  cotton',
    weight: '50 g',
    length: '80 m',
    price: 10,
  },

  {
    url: 'assets/catalogItems/Archna_Macrame_MINI.svg',
    category: 'Knitted yarn',
    name: 'Macrame MINI',
    brand: 'Arachna',
    structure: '74% merino, 26%  cotton',
    weight: '50 g',
    length: '80 m',
    price: 10,
  },

  {
    url: 'assets/catalogItems/Papyrus_Fibra_Natura.jpg',
    category: 'Knitted yarn',
    name: 'Fibra Natura',
    brand: 'Papyrus',
    structure: '100%  cotton',
    weight: '50 g',
    length: '120 m',
    price: 20,
  },

  {
    url: 'assets/catalogItems/Alize_Angora_Gold.jpg',
    category: 'Knitted yarn',
    name: 'Angora Gold',
    brand: 'Alize',
    structure: '74% merino, 26%  cotton',
    weight: '50 g',
    length: '80 m',
    price: 10,
  },

  {
    url: 'assets/catalogItems/Vento_DItalia_Angora_Rabbit.jpg',
    category: 'Knitted yarn',
    name: 'Flame',
    brand: 'Vento DItalia',
    structure: '70% fluff, 30% nylon',
    weight: '50 g',
    length: '350 m',
    price: 52,
  },
];
