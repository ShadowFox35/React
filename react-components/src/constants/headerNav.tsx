import { navElemType } from 'types/objects';

export const headerNav: navElemType[] = [
  {
    text: 'Catalog',
    link: '/',
  },
  {
    text: 'About Us',
    link: '/about',
  },
];
