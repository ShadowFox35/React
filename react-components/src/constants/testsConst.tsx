import React from 'react';
import { ReactNode } from 'react';
import { BrowserRouter } from 'react-router-dom';

export const testAppWrapper = (children: ReactNode): JSX.Element => {
  return <BrowserRouter>{children}</BrowserRouter>;
};
