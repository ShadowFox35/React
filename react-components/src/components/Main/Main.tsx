import Search from 'components/Header/Search/Search';
import React from 'react';
import Catalog from './Catalog/Catalog';

import './Main.scss';

const Main: React.FC = () => {
  document.title = 'main';
  return (
    <div className="main container">
      <Search />
      <Catalog />
    </div>
  );
};

export default Main;
