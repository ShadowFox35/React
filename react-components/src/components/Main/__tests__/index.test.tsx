import { render } from '@testing-library/react';
import { testAppWrapper } from 'constants/testsConst';
import * as React from 'react';

import Main from '../Main';

describe('<testMain />', () => {
  it('should render and match the snapshot', () => {
    const componentRender = render(testAppWrapper(<Main />));
    expect(componentRender).toMatchSnapshot();
  });
});
