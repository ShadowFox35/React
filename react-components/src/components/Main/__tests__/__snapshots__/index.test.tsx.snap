// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<testMain /> should render and match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main container"
      >
        <section
          class="search"
        >
          <div
            class="wrapper"
          >
            <input
              autocomplete="off"
              class="search_input"
              name="search"
              placeholder="Search ..."
              tabindex="0"
              value="enter text"
            />
            <div
              class="search_icon"
            >
              <img
                alt="search icon"
                class="icon"
                src="search.svg"
              />
            </div>
          </div>
        </section>
        <div
          class="catalog"
        >
          <div
            class="good"
          >
            <img
              alt="Zefirka"
              class="good_img"
              src="/assets/catalogItems/Zefirka.svg"
            />
            <div>
               
              <strong>
                Knitted yarn
              </strong>
            </div>
            <strong>
              Zefirka
            </strong>
            <div>
              100%  cotton
            </div>
            <div>
               weight: 
              380 g
            </div>
            <div>
              thread length: 
              100 m
            </div>
            <div>
               price: 
              17
               $.
            </div>
            <button
              class="good_btn-inactiv"
              type="button"
            >
              <img
                alt=""
                class="icon"
                src="basket_icon.svg"
              />
              Add
            </button>
          </div>
          <div
            class="good"
          >
            <img
              alt="Sling Print"
              class="good_img"
              src="/assets/catalogItems/Arachna.svg"
            />
            <div>
               
              <strong>
                Knitted yarn
              </strong>
            </div>
            <strong>
              Sling Print
            </strong>
            <div>
              100%  cotton
            </div>
            <div>
               weight: 
              160 g
            </div>
            <div>
              thread length: 
              50 m
            </div>
            <div>
               price: 
              16
               $.
            </div>
            <button
              class="good_btn-inactiv"
              type="button"
            >
              <img
                alt=""
                class="icon"
                src="basket_icon.svg"
              />
              Add
            </button>
          </div>
          <div
            class="good"
          >
            <img
              alt="Merino Flame"
              class="good_img"
              src="/assets/catalogItems/Katia_Merino_Flame.svg"
            />
            <div>
               
              <strong>
                Knitted yarn
              </strong>
            </div>
            <strong>
              Merino Flame
            </strong>
            <div>
              74% merino, 26%  cotton
            </div>
            <div>
               weight: 
              50 g
            </div>
            <div>
              thread length: 
              80 m
            </div>
            <div>
               price: 
              10
               $.
            </div>
            <button
              class="good_btn-inactiv"
              type="button"
            >
              <img
                alt=""
                class="icon"
                src="basket_icon.svg"
              />
              Add
            </button>
          </div>
          <div
            class="good"
          >
            <img
              alt="Macrame MAXI"
              class="good_img"
              src="/assets/catalogItems/Archna_Macrame_MAXI.svg"
            />
            <div>
               
              <strong>
                Knitted yarn
              </strong>
            </div>
            <strong>
              Macrame MAXI
            </strong>
            <div>
              74% merino, 26%  cotton
            </div>
            <div>
               weight: 
              50 g
            </div>
            <div>
              thread length: 
              80 m
            </div>
            <div>
               price: 
              10
               $.
            </div>
            <button
              class="good_btn-inactiv"
              type="button"
            >
              <img
                alt=""
                class="icon"
                src="basket_icon.svg"
              />
              Add
            </button>
          </div>
          <div
            class="good"
          >
            <img
              alt="Macrame MINI"
              class="good_img"
              src="/assets/catalogItems/Archna_Macrame_MINI.svg"
            />
            <div>
               
              <strong>
                Knitted yarn
              </strong>
            </div>
            <strong>
              Macrame MINI
            </strong>
            <div>
              74% merino, 26%  cotton
            </div>
            <div>
               weight: 
              50 g
            </div>
            <div>
              thread length: 
              80 m
            </div>
            <div>
               price: 
              10
               $.
            </div>
            <button
              class="good_btn-inactiv"
              type="button"
            >
              <img
                alt=""
                class="icon"
                src="basket_icon.svg"
              />
              Add
            </button>
          </div>
          <div
            class="good"
          >
            <img
              alt="Fibra Natura"
              class="good_img"
              src="/assets/catalogItems/Papyrus_Fibra_Natura.jpg"
            />
            <div>
               
              <strong>
                Knitted yarn
              </strong>
            </div>
            <strong>
              Fibra Natura
            </strong>
            <div>
              100%  cotton
            </div>
            <div>
               weight: 
              50 g
            </div>
            <div>
              thread length: 
              120 m
            </div>
            <div>
               price: 
              20
               $.
            </div>
            <button
              class="good_btn-inactiv"
              type="button"
            >
              <img
                alt=""
                class="icon"
                src="basket_icon.svg"
              />
              Add
            </button>
          </div>
          <div
            class="good"
          >
            <img
              alt="Angora Gold"
              class="good_img"
              src="/assets/catalogItems/Alize_Angora_Gold.jpg"
            />
            <div>
               
              <strong>
                Knitted yarn
              </strong>
            </div>
            <strong>
              Angora Gold
            </strong>
            <div>
              74% merino, 26%  cotton
            </div>
            <div>
               weight: 
              50 g
            </div>
            <div>
              thread length: 
              80 m
            </div>
            <div>
               price: 
              10
               $.
            </div>
            <button
              class="good_btn-inactiv"
              type="button"
            >
              <img
                alt=""
                class="icon"
                src="basket_icon.svg"
              />
              Add
            </button>
          </div>
          <div
            class="good"
          >
            <img
              alt="Flame"
              class="good_img"
              src="/assets/catalogItems/Vento_DItalia_Angora_Rabbit.jpg"
            />
            <div>
               
              <strong>
                Knitted yarn
              </strong>
            </div>
            <strong>
              Flame
            </strong>
            <div>
              70% fluff, 30% nylon
            </div>
            <div>
               weight: 
              50 g
            </div>
            <div>
              thread length: 
              350 m
            </div>
            <div>
               price: 
              52
               $.
            </div>
            <button
              class="good_btn-inactiv"
              type="button"
            >
              <img
                alt=""
                class="icon"
                src="basket_icon.svg"
              />
              Add
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main container"
    >
      <section
        class="search"
      >
        <div
          class="wrapper"
        >
          <input
            autocomplete="off"
            class="search_input"
            name="search"
            placeholder="Search ..."
            tabindex="0"
            value="enter text"
          />
          <div
            class="search_icon"
          >
            <img
              alt="search icon"
              class="icon"
              src="search.svg"
            />
          </div>
        </div>
      </section>
      <div
        class="catalog"
      >
        <div
          class="good"
        >
          <img
            alt="Zefirka"
            class="good_img"
            src="/assets/catalogItems/Zefirka.svg"
          />
          <div>
             
            <strong>
              Knitted yarn
            </strong>
          </div>
          <strong>
            Zefirka
          </strong>
          <div>
            100%  cotton
          </div>
          <div>
             weight: 
            380 g
          </div>
          <div>
            thread length: 
            100 m
          </div>
          <div>
             price: 
            17
             $.
          </div>
          <button
            class="good_btn-inactiv"
            type="button"
          >
            <img
              alt=""
              class="icon"
              src="basket_icon.svg"
            />
            Add
          </button>
        </div>
        <div
          class="good"
        >
          <img
            alt="Sling Print"
            class="good_img"
            src="/assets/catalogItems/Arachna.svg"
          />
          <div>
             
            <strong>
              Knitted yarn
            </strong>
          </div>
          <strong>
            Sling Print
          </strong>
          <div>
            100%  cotton
          </div>
          <div>
             weight: 
            160 g
          </div>
          <div>
            thread length: 
            50 m
          </div>
          <div>
             price: 
            16
             $.
          </div>
          <button
            class="good_btn-inactiv"
            type="button"
          >
            <img
              alt=""
              class="icon"
              src="basket_icon.svg"
            />
            Add
          </button>
        </div>
        <div
          class="good"
        >
          <img
            alt="Merino Flame"
            class="good_img"
            src="/assets/catalogItems/Katia_Merino_Flame.svg"
          />
          <div>
             
            <strong>
              Knitted yarn
            </strong>
          </div>
          <strong>
            Merino Flame
          </strong>
          <div>
            74% merino, 26%  cotton
          </div>
          <div>
             weight: 
            50 g
          </div>
          <div>
            thread length: 
            80 m
          </div>
          <div>
             price: 
            10
             $.
          </div>
          <button
            class="good_btn-inactiv"
            type="button"
          >
            <img
              alt=""
              class="icon"
              src="basket_icon.svg"
            />
            Add
          </button>
        </div>
        <div
          class="good"
        >
          <img
            alt="Macrame MAXI"
            class="good_img"
            src="/assets/catalogItems/Archna_Macrame_MAXI.svg"
          />
          <div>
             
            <strong>
              Knitted yarn
            </strong>
          </div>
          <strong>
            Macrame MAXI
          </strong>
          <div>
            74% merino, 26%  cotton
          </div>
          <div>
             weight: 
            50 g
          </div>
          <div>
            thread length: 
            80 m
          </div>
          <div>
             price: 
            10
             $.
          </div>
          <button
            class="good_btn-inactiv"
            type="button"
          >
            <img
              alt=""
              class="icon"
              src="basket_icon.svg"
            />
            Add
          </button>
        </div>
        <div
          class="good"
        >
          <img
            alt="Macrame MINI"
            class="good_img"
            src="/assets/catalogItems/Archna_Macrame_MINI.svg"
          />
          <div>
             
            <strong>
              Knitted yarn
            </strong>
          </div>
          <strong>
            Macrame MINI
          </strong>
          <div>
            74% merino, 26%  cotton
          </div>
          <div>
             weight: 
            50 g
          </div>
          <div>
            thread length: 
            80 m
          </div>
          <div>
             price: 
            10
             $.
          </div>
          <button
            class="good_btn-inactiv"
            type="button"
          >
            <img
              alt=""
              class="icon"
              src="basket_icon.svg"
            />
            Add
          </button>
        </div>
        <div
          class="good"
        >
          <img
            alt="Fibra Natura"
            class="good_img"
            src="/assets/catalogItems/Papyrus_Fibra_Natura.jpg"
          />
          <div>
             
            <strong>
              Knitted yarn
            </strong>
          </div>
          <strong>
            Fibra Natura
          </strong>
          <div>
            100%  cotton
          </div>
          <div>
             weight: 
            50 g
          </div>
          <div>
            thread length: 
            120 m
          </div>
          <div>
             price: 
            20
             $.
          </div>
          <button
            class="good_btn-inactiv"
            type="button"
          >
            <img
              alt=""
              class="icon"
              src="basket_icon.svg"
            />
            Add
          </button>
        </div>
        <div
          class="good"
        >
          <img
            alt="Angora Gold"
            class="good_img"
            src="/assets/catalogItems/Alize_Angora_Gold.jpg"
          />
          <div>
             
            <strong>
              Knitted yarn
            </strong>
          </div>
          <strong>
            Angora Gold
          </strong>
          <div>
            74% merino, 26%  cotton
          </div>
          <div>
             weight: 
            50 g
          </div>
          <div>
            thread length: 
            80 m
          </div>
          <div>
             price: 
            10
             $.
          </div>
          <button
            class="good_btn-inactiv"
            type="button"
          >
            <img
              alt=""
              class="icon"
              src="basket_icon.svg"
            />
            Add
          </button>
        </div>
        <div
          class="good"
        >
          <img
            alt="Flame"
            class="good_img"
            src="/assets/catalogItems/Vento_DItalia_Angora_Rabbit.jpg"
          />
          <div>
             
            <strong>
              Knitted yarn
            </strong>
          </div>
          <strong>
            Flame
          </strong>
          <div>
            70% fluff, 30% nylon
          </div>
          <div>
             weight: 
            50 g
          </div>
          <div>
            thread length: 
            350 m
          </div>
          <div>
             price: 
            52
             $.
          </div>
          <button
            class="good_btn-inactiv"
            type="button"
          >
            <img
              alt=""
              class="icon"
              src="basket_icon.svg"
            />
            Add
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
